register.php

```diff
@@ -0,0 +1,514 @@
+
+      // Validate Coupon Code
+      if(($coupon_code = arr_get($_POST, 'coupon_code')) != null) {
+        $coupon_price = arr_get($config['checkout']['coupons'], $coupon_code);
+        if(null == $coupon_price) {
+          $coupon_code = null;
+        }
+        else {
+          $ticket_price = $coupon_price;
+        }
+      }
+
+      // Attendee validation
+      for($i = 1; $i <= $number_of_tickets; $i++) {
+        $attendee = array(
+          'first_name' => trim(arr_get($_POST, 'first_name_' . $i, '')),
+          'last_name'  => trim(arr_get($_POST, 'last_name_' . $i, '')),
+          'email'      => trim(arr_get($_POST, 'email_' . $i, '')),
+          'twitter'    => trim(arr_get($_POST, 'twitter_' . $i, '')),
+          'company'    => trim(arr_get($_POST, 'company_' . $i, '')),
+          'job_title'  => trim(arr_get($_POST, 'job_title_' . $i, '')),
+          'shirt_size' => trim(arr_get($_POST, 'shirt_size_' . $i, '')),
+        );
+        $attendee_data[$i] = $attendee;
+
+        $errors = array();
+
+        if(empty($attendee['first_name'])) {
+          $errors['first_name'] = 'This field is required.';
+        }
+        if(empty($attendee['last_name'])) {
+          $errors['last_name'] = 'This field is required.';
+        }
+        if(! (filter_var($attendee['email'], FILTER_VALIDATE_EMAIL) && preg_match('/@.+\./', $attendee['email'])) ) {
+          $errors['email'] = 'An email is required.';
+        }
+
+        if(0 != count($errors)) {
+          $form_errors[$i] = $errors;
+        }
+      }
+
+      $receipt_email = trim(arr_get($_POST, 'receipt_email', ''));
+      if(! (filter_var($receipt_email, FILTER_VALIDATE_EMAIL) && preg_match('/@.+\./', $receipt_email)) ) {
+        $form_errors['receipt_email'] = 'An email is required.';
+      }
+
+      if(0 == count($form_errors)) {
+        $stripe_token = $_POST['stripeToken'];
+
+        $ticket_price_as_pennies = $ticket_price * 100;
+        $total_as_pennies = $ticket_price_as_pennies * $number_of_tickets;
+
+        try {
+          $charge = \Stripe\Charge::create(array(
+            "amount"        => $total_as_pennies,
+            "currency"      => "usd",
+            "source"        => $stripe_token,
+            "description"   => "$number_of_tickets NEJSCONF 2016 Tickets",
+            "receipt_email" => $receipt_email,
+            "metadata"      => array(
+              "coupon_code"       => $coupon_code,
+              "ticket_price"      => $ticket_price,
+              "number_of_tickets" => $number_of_tickets,
+            ),
+          ));
+
+					$httpAdapter = new CurlHttpAdapter();
+					$sparkpost = new SparkPost($httpAdapter, ['key' => $config['sparkpost']['api_key']]);
+
+					foreach($attendee_data as $attendee) {
+						try {
+							$results = $sparkpost->transmission->send([
+								'from'=>[
+									'name' => 'NEJS Conf',
+									'email' => 'tickets@sparkpostbox.com'
```

Comment:
Need to confirm our sending domain with sparkpost and switch this to @nejsconf.com